### PR TITLE
Fix server auth test

### PR DESCRIPTION
**Pull request by AntoinePrv**

### CODE REVIEW

File: mamba/tests/testserver.sh
Comment (AntoinePrv) [inline mamba/tests/testserver.sh:1]: Unrelated changes

File: mamba/environment-dev.yml [hunk 1]
```diff
@@ -18,9 +18,15 @@ dependencies:
...
   - fmt
+  - python
+  - pip
   - pybind11
   - pybind11-stubgen
   - pytest
   - pyyaml
   - conda
+  - conda-content-trust
+  - cryptography<40.0  # Or breaks conda-content-trust
```
Comment (AntoinePrv) [inline mamba/environment-dev.yml:29]: That's our bug

**Review by JohanMabille — APPROVED** [review]
